#!/usr/bin/env python
# set up dynamically reconfigurable parameters here

from dynamic_reconfigure.parameter_generator_catkin import \
            ParameterGenerator, double_t

PACKAGE = "twist_controller"

gen = ParameterGenerator()

gen.add("dyn_velo_proportional_control", double_t, 0, "Velocity proportional control constant", 0.2, 0, 4)
gen.add("dyn_velo_integral_control", double_t, 0, "Velocity integral control constant", 0.01, 0, 4.0)

gen.add("dyn_braking_proportional_control", double_t, 0, "Braking proportional control constant", 200, 0, 1000)
gen.add("dyn_braking_integral_control", double_t, 0, "Braking integral control constant", 0, 0, 4.0)

exit(gen.generate(PACKAGE, "twist_controller", "DynReconf"))
